sklearn/utils/estimator_checks.py

```diff
@@ -2279,6 +2274,16 @@ def enforce_estimator_tags_y(estimator, y):
     return y
 
 
+def enforce_estimator_tags_X(estimator, X):
+    # Estimators with a `requires_positive_X` tag only accept strictly positive
+    # inputs
+    if _safe_tags(estimator, "requires_positive_X"):
+        # Create strictly positive X. The minimal increment above 0 is 1, as
+        # X could be of integer dtype.
+        X += 1 + abs(X.min())
```

Comment:
why not ``X -= X.min() - 1e-5`` or something like that? I don't know why you would do abs. If your data had a minimum of ``2`` after this is has a minimum of ``5``

Reply:
I agree `X -= X.min() - 1e-5 seems better, I just had taken this line from `enforce_estimator_tags_y` above
Will do

Reply:
ahh the comment was about dtype int? hm not sure in how far that applies to X here. But if thests are passing then that's fine I guess ;) Is this ready? If so, please rename from WIP to MRG.

Reply:
Ah yes, I didn't see that comment, I guess this means if there is an integer X at one point we want to let it integer ? I don't know to what extent it applies indeed, but maybe in case we can let it with the int ? What about `X -= X.min() - 1` ? So that it'll still be ok wrt to your comment https://github.com/scikit-learn/scikit-learn/pull/14705/files/c6751b454bdf84699954e3f6bfa7000db2d42d48#r316775147 but it keeps the integer ?